addons/event_sale/views/templates.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="sale_order_portal_template" inherit_id="sale.sale_order_portal_template" name="Event sale order portal">
+        <xpath expr="//div[@t-if='sale_order.is_expired']" position="before">
+            <div t-if="sale_order.has_soldout_event" class="alert alert-warning alert-dismissible d-print-none" role="alert">
```

Comment:
As this is a qweb template (server-rendered), not a record view (client-rendered), you can just call a method. It doesn't need to be a field.

We like to avoid adding fields in stable, even non-stored ones.